collectors/python.d.plugin/isc_dhcpd/isc_dhcpd.chart.py

```diff
@@ -83,17 +83,30 @@ def get_data(self):
 
 
 class Pool:
-    def __init__(self, name, network):
+    def __init__(self, name, networks):
         self.id = re.sub(r'[:/.-]+', '_', name)
         self.name = name
-        self.network = ipaddress.ip_network(address=u'%s' % network)
+
+        self.networks = list()
+        for network_item in networks.split(" "):
+            self.networks.append( ipaddress.ip_network(address=u'%s' % network_item) )
 
     def num_hosts(self):
-        return self.network.num_addresses - 2
+        num_addres = 0
+        for network_item in self.networks:
+            num_addres += network_item.num_addresses
+            if network_item.prefixlen <= 24:
+                num_addres -= 2
 
-    def __contains__(self, item):
-        return item.address in self.network
+        return num_addres
```

Comment:
```python
            if network_item.prefixlen <= 24:
                num_addres -= 2
```

i think we can -2 for every network - subnet address + broadcast address.

something like

```python
sum([network.num_addresses -2 for network in self.networks])
```

what am i missing?

Reply:
I know about `/30`/`/31`, implying it is unlikely a person will use them as dhcp pools

Reply:
Yes, I understand why is -2
But the problem is that if we want a pool from ip 192.168.5.100 to 192.168.5.110 it will not calculate it well.
Since the CIDR addresses for that range would be:
```
192.168.0.100/30
192.168.0.104/30
192.168.0.108/31
192.168.0.110/32
```

This would give a result that there are only 3 ip's in that range, instead of 11 or 9 if we remove subnet address + broadcast address.

Without subtracting (subnet address + broadcast address):
```
Num Addres (4)
Num Addres (4)
Num Addres (2)
Num Addres (1)
Total: 11
```

Subtracting (subnet address + broadcast address):
```
Num Addres (2)
Num Addres (2)
Num Addres (0)
Num Addres (-1)
Total: 3
```

Reply:
`if network_item.prefixlen <= 24:` is not correct, i dont agree with that implementation. For CIDR it is -2, no matter what prefix length is.

if we need

> 192.168.5.100 to 192.168.5.110

then we need ranges, not CIDR.

I added them to the `dnsmasq_dhcp` module

https://github.com/netdata/go.d.plugin/blob/master/modules/dnsmasq_dhcp/ip/range.go

Reply:
simple way to add ip ranges is to implement them as `set` of `ipaddress.ip_address`